### ISSUE TITLE
Uncaught client side exception when using Grid.setFrozenColumnCount or Grid.Column.setHidden methods

### ISSUE THREAD

**Issue by alejandro-du** [vaadin/framework#8329]
The following UI implementation throws a client side exception with 8.0.0.beta1:
```
public class MyUI extends UI {

    @Override
    protected void init(VaadinRequest vaadinRequest) {
        Grid grid = new Grid();
        Grid.Column col = grid.addColumn(o -> "test");

        // any of the following throws a client-side exception:
        grid.setFrozenColumnCount(1); // breaks it
        col.setHidden(true);          // breaks it

        Button button = new Button("Click to break it", e -> setContent(new Label("See the console")));

        VerticalLayout layout = new VerticalLayout(button, grid);
        setContent(layout);
    }

    @WebServlet(urlPatterns = "/*", name = "MyUIServlet", asyncSupported = true)
    @VaadinServletConfiguration(ui = MyUI.class, productionMode = false)
    public static class MyUIServlet extends VaadinServlet {
    }
}

```
Client side exception:
```
com.vaadin.client.ApplicationConfiguration
SEVERE: The given column range (-1..0) was outside of the current number of columns (0)
java.lang.IndexOutOfBoundsException: The given column range (-1..0) was outside of the current number of columns (0)

```

The exception appears when calling `setFrozenColumnCount`, `setHidden`, or both.

See also https://vaadin.com/forum#!/thread/15034700

**Comment by carlislefox** [comment]
Can confirm this is affecting our project too on the version 8.0.0 release. In our case it is calls to 

`.setHidden(true);`

That result in it breaking.

**Comment by focbenz** [comment]
Can still reproduce the issue with the sample project using the combination of:
```
grid1.setFrozenColumnCount(1);
col1.setHidable(true);
col1.setHidden(true);
```
as soon as switch to grid2 is performed

**Comment by focbenz** [comment]
Confirmed fix with 8.0.3
